core/blockchain.go

```diff
@@ -99,15 +99,16 @@ type BlockChain struct {
 	blockCache   *lru.Cache // Cache for the most recent entire blocks
 	futureBlocks *lru.Cache // future blocks are blocks added for later processing
 
-	quit    chan struct{}
-	running int32 // running must be called automically
+	quit    chan struct{} // blockchain quit channel
+	running int32         // running must be called automically
 	// procInterrupt must be atomically called
-	procInterrupt int32 // interrupt signaler for block processing
-	wg            sync.WaitGroup
+	procInterrupt int32          // interrupt signaler for block processing
+	wg            sync.WaitGroup // chain processing wait group for shutting down
 
 	pow       pow.PoW
-	processor Processor
-	validator Validator
+	rand      *mrand.Rand
```

Comment:
I think this is a merge anomaly since @zsfelfoldi moved the rand into the header chain.